src/pudl/transform/ferc1.py

```diff
@@ -124,6 +124,117 @@ class Config:
 ################################################################################
 
 
+################################################################################
+# FERC 1 transform helper functions. Probably to be integrated into a class
+# below as methods or moved to a different module once it's clear where they belong.
+################################################################################
+def get_ferc1_dbf_rows_to_map(ferc1_engine: sa.engine.Engine) -> pd.DataFrame:
+    """Identify DBF rows that need to be mapped to XBRL columns.
+
+    Select all records in the ``f1_row_lit_tbl`` where the row literal associated with a
+    given combination of table and row number is different from the preceeding year.
+    This is the smallest set of records which we can use to reproduce the whole table by
+    expanding the time series to include all years, and forward filling the row
+    literals.
+    """
+    idx_cols = ["sched_table_name", "row_status", "row_number", "report_year"]
+    data_cols = ["row_literal"]
+    row_lit = pd.read_sql(
+        "f1_row_lit_tbl", con=ferc1_engine, columns=idx_cols + data_cols
+    ).sort_values(idx_cols)
+    row_lit["shifted"] = row_lit.groupby(
+        ["sched_table_name", "row_status", "row_number"]
+    ).row_literal.shift()
+    row_lit["changed"] = row_lit.row_literal != row_lit.shifted
+    return row_lit.loc[row_lit.changed, idx_cols + data_cols]
+
+
+def update_ferc1_dbf_xbrl_glue(ferc1_engine: sa.engine.Engine) -> pd.DataFrame:
+    """Regenerate the FERC 1 DBF+XBRL glue while retaining existing mappings.
+
+    Reads all rows that need to be mapped out of the ``f1_row_lit_tbl`` and
+    appends columns containing any previously mapped values, returning the
+    resulting dataframe.
+
+    This
+    """
+    idx_cols = ["sched_table_name", "row_status", "row_number", "report_year"]
+    all_rows = get_ferc1_dbf_rows_to_map(ferc1_engine).set_index(idx_cols)
+    with importlib.resources.open_text(
+        "pudl.package_data.ferc1", "dbf_to_xbrl.csv"
+    ) as file:
+        mapped_rows = (
+            pd.read_csv(file).set_index(idx_cols).drop(["row_literal"], axis="columns")
+        )
```

Comment:
I don't think so. This function is reading the "mapped" data out of the saved file, and slapping it onto a fresh copy of **all** the (similarly indexed) information we want out of the database, some of which is part of the index, and some of which may only be there to help the humans doing the pattern matching, or to enable us to fill in the gaps. I drop `row_literal` so that there's only one copy of the column (the one coming from the DB) when the two dataframes are concatenated.